vis_sink.py

```diff
@@ -0,0 +1,374 @@
+
+    def set_params(self, params):
+        pass
+
+    def process(self):
+        self.current_time_for_log = time.time()
+        if self.options.arm_right:
+            rarm_qin_bottle = self.rarm_qin_port.read(False)
+            if rarm_qin_bottle:
+                rarm_qin = list(
+                    map(yarp.Value.asDouble,
+                        list(
+                            map(rarm_qin_bottle.get,
+                                list(range(rarm_qin_bottle.size()))))))
+                self.add_log({"id": "arm_right", "set_angles": rarm_qin})
+        if self.options.arm_left:
+            larm_qin_bottle = self.larm_qin_port.read(False)
+            if larm_qin_bottle:
+                larm_qin = list(
+                    map(yarp.Value.asDouble,
+                        list(
+                            map(larm_qin_bottle.get,
+                                list(range(larm_qin_bottle.size()))))))
+                self.add_log({"id": "arm_right", "set_angles": larm_qin})
+        if self.options.hand_right:
+            self.get_hand_info(self.right_hand_port, "right")
+
+        if self.options.hand_left:
+            self.get_hand_info(self.left_hand_port, "left")
+
+        # objects port processing
+        while self.objects_port.getPendingReads() != 0:
+            objects_bottle = self.objects_port.read(True)
+            if objects_bottle:
+                objects_bottles = list(
+                    map(yarp.Value.asList,
+                        list(
+                            map(objects_bottle.get,
+                                list(range(objects_bottle.size()))))))
+                for object_bottle in objects_bottles:
+                    object_id = object_bottle.get(0).asInt()
+                    object_prop_name = object_bottle.get(1).toString()
+                    if object_prop_name == 'type':
+                        object_prop_data = object_bottle.get(2).toString()
+                    else:
+                        object_prop_data = array(
+                            list(
+                                map(yarp.Value.asDouble,
+                                    list(
+                                        map(object_bottle.get,
+                                            list(range(
+                                                object_bottle.size()))[2:])))))
+                    # Properties are: type (sphere, frame, arrow, box), pose
+                    # (16 values), color (three values), scale (three values),
+                    # etc
+                    self.add_log({
+                        "id": object_id,
+                        "object_prop_name": object_prop_name,
+                        "object_prop_data": object_prop_data
+                    })
+
+    def get_hand_info(self, hand_port, side_str):
+        angles_field = list(range(3))
+        speeds_field = list(range(3, 6))
+        torques_field = list(range(6, 9))
+        THUMB_ANGLE_FIELD = 1
```

Comment:
An enum might be better [docs](https://docs.python.org/3/library/enum.html), but please include this module as depedency if you do [https://pypi.python.org/pypi/enum34](https://pypi.python.org/pypi/enum34) since the enum is only part of the standard library from python 3.4 and onwards

Reply:
hmmmm.... This should work in python2.

Reply:
Yes that is why I suggest `enum34` package it will work on python2, it is a backport of the `Enum`  module in the standard package to older python versions. 

You can use Enum in older version of python if you install `enum34` package (`pip install enum34`). Just remember to add `enum34` as requirement, since it is easy to forget if you are developing on python3.4 and above